services/tokei/tokei-github-lines.service.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  lines: nonNegativeInteger,
+}).required()
+
+const documentation = `
+<p>
+  The <code>provider</code> is the domain name of git host.
+  If no TLD is provided, <code>.com</code> will be added.
+  For example, setting <code>gitlab</code> or <code>bitbucket.org</code> as the provider also works.
+</p>
+`
+
+module.exports = class LinesOfCode extends BaseJsonService {
+  static category = 'size'
+
+  static route = { base: 'tokei/l', pattern: ':provider/:user/:repo' }
+
+  static get examples() {
+    return [
+      {
+        title: 'Lines of code',
+        namedParams: {
+          provider: 'github',
+          user: 'badges',
+          repo: 'shields',
+        },
+        staticPreview: this.render({ lines: 119500 }),
+        keywords: ['loc', 'tokei'],
+        documentation,
+      },
+    ]
+  }
+
+  static defaultBadgeData = { label: 'total lines' }
+
+  static render({ lines }) {
+    return {
+      label: 'total lines',
+      message: metric(lines),
+      color: 'brightgreen',
+    }
+  }
+
+  async fetch({ provider, user, repo }) {
+    return this._requestJson({
+      schema,
+      url: `https://tokei.rs/b1/${provider}/${user}/${repo}`,
```

Comment:
I think we should add a comment here to speak to the API behavior. All the info on the [tokei_rs](https://github.com/XAMPPRocky/tokei_rs) GitHub page speaks to badges, but it will return json provided the proper request headers.

This isn't a [violation of our guidelines](https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines) or anything (at least not in my mind) but will be helpful for future observers. Will also be helpful to have a link to the tokei_rs repo since that's where the code for the endpoint is hosted

Reply:
Sure, will do

Reply:
I'll also link [this](https://github.com/XAMPPRocky/tokei_rs/issues/8#issuecomment-475071147) in the comment.

Reply:
Done.